ledger/accountdb_test.go

```diff
@@ -3313,3 +3323,39 @@ func TestAccountOnlineAccountsNewRound(t *testing.T) {
 	_, _, err = onlineAccountsNewRoundImpl(writer, updates, proto, lastUpdateRound)
 	require.Error(t, err)
 }
+
+func TestAccountOnlineRoundParams(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	proto := config.Consensus[protocol.ConsensusCurrentVersion]
+
+	dbs, _ := dbOpenTest(t, true)
+	setDbLogging(t, dbs)
+	defer dbs.Close()
+
+	tx, err := dbs.Wdb.Handle.Begin()
+	require.NoError(t, err)
+	defer tx.Rollback()
+
+	var accts map[basics.Address]basics.AccountData
+	accountsInitTest(t, tx, accts, protocol.ConsensusCurrentVersion)
+
+	onlineRoundParams := make([]ledgercore.OnlineRoundParamsData, 80+proto.MaxBalLookback)
+	for i := range onlineRoundParams {
+		onlineRoundParams[i].OnlineSupply = uint64(i)
+	}
+
+	err = accountsPutOnlineRoundParams(tx, onlineRoundParams, 1)
+	require.NoError(t, err)
+
+	dbOnlineRoundParams, err := accountsOnlineRoundParams(tx)
+	require.NoError(t, err)
+	require.Equal(t, onlineRoundParams, dbOnlineRoundParams[1:])
```

Comment:
not sure I understand: `accountsPutOnlineRoundParams` inserts all `80+proto.MaxBalLookback` entries into a table with the first round = 1. Then `accountsOnlineRoundParams` read everything. So why `onlineRoundParams` and `dbOnlineRoundParams` are not equal after the first read?

Reply:
Because the db is initalized with an entry for round 0 at `accountsInitTest`

Reply:
then please add an assert `len(dbOnlineRoundParams ) == 80+proto.MaxBalLookback + 1` and a comment +1 is from the init state.